File: src/main/java/land/leets/domain/comment/domain/Comment.java
package land.leets.domain.comment.domain;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToOne;
import land.leets.domain.admin.domain.Admin;
import land.leets.domain.shared.BaseTimeEntity;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity(name = "comments")
@Builder
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
public class Comment extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long applicationId;

    @Column(nullable = false)
    private String content;

    @OneToOne
    private Admin admin;
}

File: src/main/kotlin/land/leets/domain/comment/domain/Comment.kt
package land.leets.domain.comment.domain

import jakarta.persistence.*
import land.leets.domain.admin.domain.Admin
import land.leets.domain.shared.BaseTimeEntity

@Entity(name = "comments")
class Comment(
    @Column(nullable = false)
    val applicationId: Long,

    @Column(nullable = false)
    val content: String,

    @ManyToOne(fetch = FetchType.LAZY)
    val admin: Admin,

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L
) : BaseTimeEntity()
